src/inference/src/dev/icompiled_model.cpp

```diff
@@ -0,0 +1,49 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/runtime/icompiled_model.hpp"
+
+#include "cpp_interfaces/interface/ie_iexecutable_network_internal.hpp"
+#include "icompiled_model_wrapper.hpp"
+#include "openvino/core/model.hpp"
+
+ov::ICompiledModel::ICompiledModel(const std::shared_ptr<const ov::Model>& model,
+                                   const std::shared_ptr<const ov::IPlugin>& plugin,
+                                   const InferenceEngine::ITaskExecutor::Ptr& task_executor,
+                                   const InferenceEngine::ITaskExecutor::Ptr& callback_executor)
+    : m_plugin(plugin),
+      m_task_executor(task_executor),
+      m_callback_executor(callback_executor) {
+    if (model) {
```

Comment:
do we need this check here? Maybe in case of import / export we need a different ctor?